src/main/java/com/dimensiondelvers/dimensiondelvers/datagen/ModLanguageProvider.java

```diff
@@ -28,10 +31,26 @@ protected void addTranslations() {
         addItem(ModItems.EXAMPLE_ITEM, "Example Item");
         addItem(ModItems.RUNEGEM, "Runegem");
 
+        ModBlocks.BLOCK_FAMILY_HELPERS.forEach(helper -> {
+            addBlock(helper.getBlock(), getTranslationString(helper.getBlock().get()));
+            helper.getVariants().forEach((variant, block) -> addBlock(block, getTranslationString(block.get())));
+        });
+
 
         // Adds a generic translation
         add("itemGroup.dimensiondelvers", "Dimension Delvers");
 
         add("container.dimensiondelvers.rune_anvil", "Rune Anvil");
     }
+
+    private static @NotNull String getTranslationString(Block block) {
+        String idString = BuiltInRegistries.BLOCK.getKey(block).getPath();
+        StringBuilder sb = new StringBuilder();
+        for (String word : idString.toLowerCase().split("_")) {
```

Comment:
I recommend specifying a Locale when using toLowerCase (Probably `Locale.ROOT`).

Reply:
Same for toUpperCase below

Reply:
Done